src/main/scala/org/marvin/executor/api/GenericAPI.scala

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright [2017] [B2W Digital]
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.marvin.executor.api
+
+import java.util.concurrent.Executors
+
+import actions.HealthCheckResponse.Status
+import akka.Done
+import akka.actor.{ActorRef, ActorSystem}
+import akka.dispatch.MessageDispatcher
+import akka.event.{Logging, LoggingAdapter}
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.{ContentTypes, HttpEntity, HttpResponse, StatusCodes}
+import akka.http.scaladsl.server.{HttpApp, Route, StandardRoute}
+import akka.pattern.ask
+import akka.util.Timeout
+import org.marvin.executor.actions.BatchAction.{BatchExecute, BatchHealthCheck, BatchReload}
+import org.marvin.executor.actions.OnlineAction.{OnlineExecute, OnlineHealthCheck}
+import org.marvin.executor.actions.PipelineAction.PipelineExecute
+import org.marvin.executor.api.GenericAPI.{DefaultBatchRequest, DefaultHttpResponse, DefaultOnlineRequest, HealthStatus}
+import org.marvin.executor.statemachine.Reload
+import org.marvin.model.EngineMetadata
+import org.marvin.util.ProtocolUtil
+import spray.json.{DefaultJsonProtocol, RootJsonFormat}
+
+import scala.concurrent.duration._
+import scala.concurrent._
+import scala.util.{Failure, Success, Try}
+
+trait GenericAPIFunctions {
```

Comment:
[minor] the suffix Functions looks redundant for me. It's like naming a java interface like FooInterface

Reply:
Any suggestion?

Remembering that this trait (GenericAPIFunctions) was created not because we need it as a kind of generalisation but cause we need to mock it .... in our unit tests...